core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -156,47 +163,102 @@ object ConsumerGroupCommand {
       val props = if (opts.options.has(opts.commandConfigOpt)) Utils.loadProps(opts.options.valueOf(opts.commandConfigOpt)) else new Properties()
       val channelSocketTimeoutMs = props.getProperty("channelSocketTimeoutMs", "600").toInt
       val channelRetryBackoffMs = props.getProperty("channelRetryBackoffMsOpt", "300").toInt
+      if (!zkUtils.getConsumerGroups().contains(group)) {
+        println("The consumer group provided does not exist")
+        return
+      }
       val topics = zkUtils.getTopicsByConsumerGroup(group)
       if (topics.isEmpty)
         println("No topic available for consumer group provided")
       printDescribeHeader()
-      topics.foreach(topic => describeTopic(group, topic, channelSocketTimeoutMs, channelRetryBackoffMs))
-    }
 
-    private def describeTopic(group: String,
-                              topic: String,
-                              channelSocketTimeoutMs: Int,
-                              channelRetryBackoffMs: Int) {
-      val topicPartitions = getTopicPartitions(topic)
-      val groupDirs = new ZKGroupTopicDirs(group, topic)
-      val ownerByTopicPartition = topicPartitions.flatMap { topicPartition =>
-        zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)._1.map { owner =>
-          topicPartition -> owner
+      val topicPartitions = getTopicsPartitions(topics)
+      var groupMemberIds = zkUtils.getConsumersInGroup(group)
+
+      // mapping of ephemeral owner -> owner
+      val ownerByEOwner = scala.collection.mutable.Map[Long, String]()
+      // mapping of topic partition -> ephemeral owner
+      val eOwnerByTopicPartition = scala.collection.mutable.Map[TopicAndPartition, Long]()
+      // mapping of topic partition -> owner
+      val ownerByTopicPartition = scala.collection.mutable.Map[TopicAndPartition, String]()
```

Comment:
Reporting both owner and member-id can be a bit confusing. Also, for ZK based consumer, we get the following output. The member-id part is repeated in the owner part. Perhaps instead, we can report 3 fields: member-id, client-id, and client-ip. For Kafka-based consumer, we can fill in all 3 fields. For ZK-based consumer, we can just fill in member-id. We can leave the client-id and client-ip part empty since that are not stored explicitly. 

```
GROUP                          TOPIC                          PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             OWNER                                         MEMBER-ID
test-consumer-group            test                           0          2               2               0               test-consumer-group_Juns-MacBook-Pro.local-1466121481202-cb9a4849-0 test-consumer-group_Juns-MacBook-Pro.local-1466121481202-cb9a4849
test-consumer-group            test                                                                                                                                    test-consumer-gr
```